File: core-tests/jdk-8/spring-rest-openapi-v2-tests/src/test/java/org/evomaster/e2etests/spring/examples/expectations/ExpectationsEMTest.java
package org.evomaster.e2etests.spring.examples.expectations;

import com.foo.rest.examples.spring.expectations.ExpectationsController;
import org.evomaster.core.problem.rest.data.HttpVerb;
import org.evomaster.core.problem.rest.data.RestIndividual;
import org.evomaster.core.search.Solution;
import org.evomaster.e2etests.spring.examples.SpringTestBase;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertTrue;

public class ExpectationsEMTest extends SpringTestBase {

    @BeforeAll
    public static void initClass() throws Exception{

        SpringTestBase.initClass(new ExpectationsController());
    }

    @Test
    public void testRunEM() throws Throwable{

        runTestHandlingFlakyAndCompilation(
                "ExpectationsEM",
                "org.bar.ExpectationsEM",
                10_000,
                (args) -> {
                    Solution<RestIndividual> solution = initAndRun(args);

                    assertTrue(solution.getIndividuals().size() >= 1);

                    assertHasAtLeastOne(solution, HttpVerb.GET, 200, "/api/basicResponsesNumeric/{value}", "42");

                    assertHasAtLeastOne(solution, HttpVerb.GET, 500, "/api/basicResponsesNumeric/{value}", "");
                }
        );
    }
}

File: e2e-tests/spring-rest-openapi-v3/src/test/kotlin/org/evomaster/e2etests/spring/openapi/v3/expectations/ExpectationsEMTest.kt
package org.evomaster.e2etests.spring.openapi.v3.examples.expectations

import com.foo.rest.examples.spring.openapi.v3.expectations.ExpectationsController
import io.restassured.RestAssured.given
import org.evomaster.core.problem.rest.HttpVerb
import org.evomaster.e2etests.spring.openapi.v3.SpringTestBase
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test

class ExpectationsEMTest : SpringTestBase() {
    //TODO: BMR - more expectations related tests are possible. This is a good place for them.

    companion object {
        @BeforeAll
        @JvmStatic
        fun init() {
            initClass(ExpectationsController())
        }
    }

    @Test
    fun testRunManual(){
        given().accept("*/*")
                .get(baseUrlOfSut + "/api/expectations/" + true)
                .then()
                .statusCode(200)

        given().accept("*/*")
                .get(baseUrlOfSut + "/api/expectations/" + false)
                .then()
                .statusCode(500)

    }

    @Test
    fun testRunEM(){
        runTestHandlingFlakyAndCompilation(
                "ExpectationsEM",
                "org.foo.ExpectationsEM",
                100
        ){args: MutableList<String> ->
            args.add("--expectationsActive")
            args.add("" + true)
            args.add("--testSuiteSplitType")
            args.add("NONE")

            val solution = initAndRun(args)

            assertTrue(solution.individuals.size >= 1)
            assertHasAtLeastOne(solution, HttpVerb.GET, 200)
            assertHasAtLeastOne(solution, HttpVerb.GET, 500)
        }
    }
}